File: 3612-adjacent-increasing-subarrays-detection-i/3612-adjacent-increasing-subarrays-detection-i.java
class Solution {
    public boolean hasIncreasingSubarrays(List<Integer> nums, int k) {
        int len=nums.size();
        int adj=0;
        for(int j=0;j<len;j++){
            adj=0;
        for(int i=j;i<len;i+=k){
            if(check(nums,i,k)){
                adj+=1;
                        System.out.println(adj);

                if(adj==2){return true;}
            }
            else{break;}            
        }}
        System.out.println(adj);
        return false;
    }

    public boolean check(List<Integer> nums, int st,int k){
        if(st+k>nums.size()){return false;}
        int len=st+k;
        for(int i=st;i<len-1;i++){
            if(nums.get(i)>=nums.get(i+1)){
                return false;
            }
        }
        return true;
    }
}

File: 3612-adjacent-increasing-subarrays-detection-i/3612-adjacent-increasing-subarrays-detection-i.cs
public class Solution {
    public bool HasIncreasingSubarrays(IList<int> nums, int k) {
        int length=nums.Count;
        for(int i=0;i+2*k<=length;i++){

            if(Check(nums,i,i+k) && Check(nums,i+k,i+2*k)){
                return true;
            }
        }
        return false;
    }

    public bool Check(IList<int> nums, int start,int end){
        
        for(int i=start;i<end-1;i++){
            if(nums[i]>=nums[i+1]){return false;}
        }
        return true;
    }
}